src/main/java/frc/robot/classes/FieldConstants.java

```diff
@@ -242,12 +242,48 @@ public static final class LoadingZone {
               Units.inchesToMeters(18.22),
               new Rotation3d()));
 
+  public static enum NodePosition{
+    MidLeft(false, true),
+    MidRight(false, false),
+    HighLeft(true, true),
+    HighRight(true, false);
+
+    private Translation2d[] rowArray;
+    private boolean isLeft;
+
+    private NodePosition(boolean isHigh, boolean isLeft){
+      this.rowArray = isHigh ? Grids.highTranslations : Grids.midTranslations;
+      this.isLeft = isLeft;
+    }
+
+    public Translation2d getTranslation(int aprilTagID){
+      Translation2d translation;
+      switch (aprilTagID){
+        case 1:
+        case 8:
+          translation = isLeft ? rowArray[1] : rowArray[3];
+          break;
+        case 7:
+        case 2:
+          translation = isLeft ? rowArray[4] : rowArray[6];
+          break;
+        case 3:
+        case 6:
+          translation = isLeft ? rowArray[7] : rowArray[9];
+          break;
+        default:
+          translation = null;
+          break;
+        }
+      return FieldConstants.fieldElementFlip(translation);
+    }
+  }
   /**
    * Flips a translation to the correct side of the field based on the current alliance color. By
    * default, all translations and poses in {@link FieldConstants} are stored with the origin at the
    * rightmost point on the BLUE ALLIANCE wall.
    */
-  public static Translation2d allianceFlip(Translation2d translation) {
+  public static Translation2d fieldElementFlip(Translation2d translation) {
     if (DriverStation.getAlliance() == Alliance.Red) {
```

Comment:
Should we use Robot.isRed?  To be consistent?  ( Realize that this is not a new change, just noticed it. )